docs/plugins/data/query-editor-enhancements.md

```diff
@@ -33,7 +33,7 @@ export interface QueryEditorExtensionConfig {
    * A function that determines if the query editor extension is enabled and should be rendered on UI.
    * @returns whether the extension is enabled.
    */
-  isEnabled$: (dependencies: QueryEditorExtensionDependencies) => Observable<boolean>;
+  isEnabled$: (dependencies: QueryEditorExtensionDependencies) => Promise<Observable<boolean>> | Observable<boolean>;
```

Comment:
do we still need `Promise<Observable<boolean>>`? i feel it's a bit anti pattern, it should either be promise or observable, but an observable promise feels weird

Reply:
hmm i see your point, let me remove that here